unity/Assets/Scripts/AgentManager.cs

```diff
@@ -348,28 +348,30 @@ private enum serverTypes { WSGI, FIFO };
         get => this.primaryAgent;
     }
 
-    private IEnumerator addAgents(ServerAction action) {
-        yield return null;
-        Vector3[] reachablePositions = primaryAgent.getReachablePositions(2.0f);
-        for (int i = 1; i < action.agentCount && this.agents.Count < Math.Min(agentColors.Length, action.agentCount); i++) {
-            action.x = reachablePositions[i + 4].x;
-            action.y = reachablePositions[i + 4].y;
-            action.z = reachablePositions[i + 4].z;
-            addAgent(action);
-            yield return null; // must do this so we wait a frame so that when we CapsuleCast we see the most recently added agent
-        }
-        for (int i = 0; i < this.agents.Count; i++) {
-            this.agents[i].m_Camera.depth = 1;
-        }
-        this.agents[0].m_Camera.depth = 9999;
+    private void addAgents(ServerAction action) {
+        if (action.agentCount > 1) {
```

Comment:
Why do we change it to a non-async function? what impact does this have?

Reply:
I think the async functionality was there just to ensure that the transforms' positions/rotations were updated as we do some collision checks in `addAgents`. I believe this should be solved by using `Physics.SyncTransforms()` rather than waiting for a frame but let me know if not.

Reply:
Ok cool, yeah seems like that was the only purpose to wait till the end of frame this looks good.